mars/services/task/execution/ray/executor.py

```diff
@@ -519,26 +627,41 @@ async def cancel(self):
         await asyncio.gather(*to_be_cancelled_coros)
 
     async def _load_subtask_inputs(
-        self, stage_id: str, subtask: Subtask, chunk_graph: ChunkGraph, context: Dict
+        self,
+        stage_id: str,
+        subtask: Subtask,
+        context: Dict,
+        shuffle_manager: ShuffleManager,
     ):
         """
-        Load a dict of input key to object ref of subtask from context.
+        Load input object refs of subtask from context.
 
         It updates the context if the input object refs are fetched from
         the meta service.
         """
-        key_to_input = {}
+        input_object_refs = []
         key_to_get_meta = {}
-        chunk_key_to_data_keys = get_chunk_key_to_data_keys(chunk_graph)
-        for key, _ in iter_input_data_keys(
-            subtask, chunk_graph, chunk_key_to_data_keys
-        ):
-            if key in context:
-                key_to_input[key] = context[key]
-            else:
-                key_to_get_meta[key] = self._meta_api.get_chunk_meta.delay(
-                    key, fields=["object_refs"]
-                )
+        # for non-shuffle chunks, chunk key will be used for indexing object refs.
+        # for shuffle chunks, mapper subtasks will have only one mapper chunk, and all outputs for mapper
+        # subtask will be shuffle blocks, the downstream reducers will receive inputs in the mappers order.
+        start_chunks = list(subtask.chunk_graph.iter_indep())
+        for index, start_chunk in enumerate(start_chunks):
+            if isinstance(start_chunk.op, Fetch):
+                # don't skip `pure_depend_keys`, otherwise pure_depend subtask execution order can't be ensured,
+                # since those ray tasks don't has dependencies on other subtasks.
```

Comment:
All the pure dependants are yield and executed when tiling (the iterative tiling), can you provide some cases that the execution is out of order?

Reply:
The comments is stale, I'll remove it.

Reply:
Changed to return None for pure_depdend_keys instead of coresponding ObjectRef